go/tools/gazelle/packages/fileinfo.go

```diff
@@ -157,6 +138,25 @@ func fileNameInfo(dir, name string) fileInfo {
 		category = ignoredExt
 	}
 
+	// Determine test, goos, and goarch. This is intended to match the logic
+	// in goodOSArchFile in go/build.
+	var isTest bool
+	var goos, goarch string
+	l := strings.Split(name[:len(name)-len(ext)], "_")
+	if len(l) >= 2 && l[len(l)-1] == "test" {
+		isTest = true && category != unsupportedExt && category != ignoredExt
```

Comment:
Not sure this is the right test, this would still allow any other category through, I think it should check that the category is exactly the go one.
I also think it would be easier to make the much smaller change of 
```isTest:   isTest && category == goExt```
down below
I'm not even convinced that the comment is how things should be, I think we should allow non go files to be marked as test and the go_test rule should be adding only files with the go category, but I will leave that decision to Jay